src/main/java/net/starlark/java/eval/Starlark.java

```diff
@@ -148,7 +148,11 @@ public static boolean isImmutable(Object x) {
    * @throws EvalException otherwise.
    */
   public static void checkHashable(Object x) throws EvalException {
-    if (x instanceof StarlarkValue) {
+    if (x instanceof String) {
+      // Strings are the most common dict keys.
+      // Check them first for that reason, but also because
+      // instanceof interface if very slow.
```

Comment:
Please add an explicit return to make it clear what this branch is doing.

Nit: "is slow", not "if very slow"